protocols/smtp/src/main/java/org/apache/james/protocols/smtp/core/fastfail/SpamTrapHandler.java

```diff
@@ -42,13 +45,20 @@ public class SpamTrapHandler implements RcptHook {
     private static final Logger LOGGER = LoggerFactory.getLogger(SpamTrapHandler.class);
 
     /** Map which hold blockedIps and blockTime in memory */
-    private final Map<String,Long> blockedIps = new HashMap<>();
+    private final Map<String,Long> blockedIps;
+    private final Clock clock;
     
     private Collection<String> spamTrapRecips = new ArrayList<>();
     
     /** Default blocktime 12 hours */
     protected long blockTime = 4320000;
 
+    @Inject
+    public SpamTrapHandler(Clock clock) {
```

Comment:
This would beak Spring injection.

Please put two constructor: @Inject on a constructor with not arguments relying on system clock, and an other one to specify the clock in tests.